api/v1alpha4/virtualmachine_types.go

```diff
@@ -793,6 +794,29 @@ type VirtualMachineSpec struct {
 	//
 	// Defaults to Online.
 	PromoteDisksMode VirtualMachinePromoteDisksMode `json:"promoteDisksMode,omitempty"`
+
+	// +optional
+
+	// CurrentSnapshot represents the desired snapshot that the
+	// virtual machine should point to. This field can have three
+	// possible values:
+	//
+	// The value of this field is nil when the working snapshot is at
+	// the root of the snapshot tree.
+	//
+	// When a new snapshot is requested by creating a new
+	// VirtualMachineSnapshot, the value of this field is set to the
+	// new snapshot resource's name.
+	//
+	// If the value of this field is set to an existing snapshot that
+	// is different from the status.currentSnapshot, the virtual machine is
+	// reverted to the requested snapshot. A revert operation results
+	// in the virtual machine's configuration and data being rolled
+	// back to the state of the snapshot. When other fields of the
+	// VirtualMachineSpec (e.g., PowerState) are updated along with
+	// the revert operation, then those are applied once the revert is
+	// completed.
+	CurrentSnapshot *corev1.LocalObjectReference `json:"currentSnapshot,omitempty"`
```

Comment:
Do we wanna make the spec.CurrentSnapshot and status.CurrentSnapshot have same types? for easier comparison.